Sources/Valet/SecureEnclaveValet.swift

```diff
@@ -226,13 +246,78 @@ public final class SecureEnclaveValet: NSObject {
 
     // MARK: Internal Properties
 
+    /// A provider of an `LAContext` used when needing to manually evaluate an authentication policy
+    /// prior to retrieving a value from the keychain in order to allow the evaluation prompt to show a
+    /// custom fallback button to the user.  Should always return a new instance of `LAContext` in
+    /// practice, but specified as an internal var to allow it to be overridden in tests.
+    internal var authenticationContextProvider: () -> LAContext = { LAContext() }
     internal let service: Service
 
     // MARK: Private Properties
 
     private let lock = NSLock()
     private let baseKeychainQuery: [String : AnyHashable]
 
+    // MARK: - Private Methods
+
+    /// Synchronously calls `LAContext.evaluatePolicy`, passing the evaluated context
+    /// via `kSecUseAuthenticationContext` when querying for the string at the given
+    /// key in the secure enclave.
+    private func synchronouslyEvaluatePolicy(
+        forKey key: String,
+        withPrompt userPrompt: String,
+        withFallbackTitle fallbackTitle: String
+    ) throws -> String {
+        // Use a semaphore to block the thread and perform the evaluation synchronously.
+        let semaphore = DispatchSemaphore(value: 0)
+        var result: Result<String, Error>?
+
+        let authenticationContext = authenticationContextProvider()
+        authenticationContext.localizedFallbackTitle = fallbackTitle
+        authenticationContext.evaluatePolicy(
+            accessControl.policy,
+            localizedReason: userPrompt,
+            reply: { [unowned self] success, error in
```

Comment:
Why not capture `self`? Won't this block always be destroyed after the prompt completes? At which point the retain cycle is both temporary and safe.

Reply:
Yeah true, I'll update